File: MarkLite/Controller/MenuViewController.m
//
//  MenuViewController.m
//  MarkLite
//
//  Created by zhubch on 15-3-27.
//  Copyright (c) 2015年 zhubch. All rights reserved.
//

#import "MenuViewController.h"
#import "Configure.h"
#import "AboutViewController.h"
#import "StyleViewController.h"
#import "ImageViewController.h"

@interface MenuViewController ()

@end

@implementation MenuViewController
{
    NSArray *items;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    

    self.navigationItem.leftBarButtonItem.tintColor = [UIColor whiteColor];
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc]initWithTitle:ZHLS(@"Back") style:UIBarButtonItemStylePlain target:self action:@selector(back)];
    
    if (kDevicePad) {
        items = @[
                    @[@"ImageResolution"],
                    @[@"AssistKeyboard",@"Font",@"Style"],
                    @[@"RateIt",@"Feedback"],
                    @[@"About"]
                ];
    }else{
        items = @[
                    @[@"ImageResolution"],
                    @[@"AssistKeyboard",@"Style"],
                    @[@"RateIt",@"Feedback"],
                    @[@"About"]
                  ];
    }
    
}

- (void)viewWillAppear:(BOOL)animated
{
    self.tabBarController.title = ZHLS(@"NavTitleOptions");
}

- (void)back {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)switchKeyboard:(UISwitch*)s{
    [Configure sharedConfigure].keyboardAssist = s.on;
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return items.count;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [items[section] count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = nil;
    
    NSString *title = items[indexPath.section][indexPath.row];

    if ([title isEqualToString:@"AssistKeyboard"]) {
        cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:@""];
        UISwitch *s = [[UISwitch alloc]initWithFrame:CGRectMake(self.view.bounds.size.width - 60, 7, 0, 0)];
        s.on = [Configure sharedConfigure].keyboardAssist;
        [s addTarget:self action:@selector(switchKeyboard:) forControlEvents:UIControlEventValueChanged];
        [cell addSubview:s];
    }else{
        cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@""];
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    }
    
    cell.textLabel.text = ZHLS(title);

    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 44;
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
{
    return 15;
}

- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
{
    return 0.1;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSDictionary *dic = @{
                          @"ImageResolution":@"resolution",
                          @"AssistKeyboard":@"",
                          @"Font":@"font",
                          @"Style":@"style",
                          @"RateIt":@"rate",
                          @"Feedback":@"feedback",
                          @"About":@"about",};
    NSString *key = items[indexPath.section][indexPath.row];

    if ([dic[key] length] > 0) {
        SEL selector = NSSelectorFromString(dic[key]);
        IMP imp = [self methodForSelector:selector];
        void (*func)(id, SEL) = (void *)imp;
        func(self, selector);
    }
}

- (void)resolution
{
    UIViewController *vc = [[ImageViewController alloc]init];
    vc.title = ZHLS(@"ImageResolution");
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)font
{
    [self performSegueWithIdentifier:@"font" sender:self];
}

- (void)style
{
    UIViewController *vc = [[StyleViewController alloc]init];
    vc.title = ZHLS(@"Style");
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)rate
{
    [[UIApplication sharedApplication]openURL:[NSURL URLWithString:@"http://itunes.apple.com/WebObjects/MZStore.woa/wa/viewContentsUserReviews?id=1098107145&pageNumber=0&sortOrdering=2&type=Purple+Software&mt=8"]];
}

- (void)feedback
{
    NSString *url = @"mailto:cheng4741@gmail.com?subject=MarkLite%20Report&body=";
    [[UIApplication sharedApplication] openURL: [NSURL URLWithString: url]];
}

- (void)about
{
    UIViewController *vc = [[AboutViewController alloc]init];
    vc.title = ZHLS(@"About");
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: MarkLite/ViewController/MenuViewController.swift
//
//  MenuViewController.swift
//  MarkLite
//
//  Created by zhubch on 2017/6/23.
//  Copyright © 2017年 zhubch. All rights reserved.
//

import UIKit
import SideMenu

class MenuViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    @IBOutlet weak var tableView: UITableView! {
        didSet {
            tableView.rowHeight = 40
        }
    }
    
    let items = [
        ("同步",[
            ("UseCloud",#selector(icloud(_:)))]),
        ("功能",[
            ("PicServer",#selector(picServer)),
            ("Font",#selector(font)),
            ("Style",#selector(style))
            ]),
        ("支持一下",[
            ("RateIt",#selector(rate)),
            ("Donate",#selector(donate))
            ]),
        ("问题反馈",[
            ("Feedback",#selector(feedback))
            ])
        
    ]
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "设置"
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return items.count
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return items[section].1.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "settings", for: indexPath)
        let item = items[indexPath.section].1[indexPath.row]
        cell.textLabel?.text = item.0
        return cell
    }

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let item = items[indexPath.section].1[indexPath.row]
        perform(item.1)
    }
    
    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
        let label = UILabel(x: 0, y: 0, w: windowWidth, h: 20)
        label.text = "  " + items[section].0
        label.textColor = rgb("a0a0a0")
        label.font = UIFont.font(ofSize: 12)
        return label
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 20
    }
    
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 10
    }
}

extension MenuViewController {
    
    func icloud(_ sender: UISwitch) {
        
    }
    
    func rate() {
        
    }
    
    func donate() {
        
    }
    
    func feedback() {
        
    }
    
    func font() {
        
    }
    
    func style() {
        
    }
    
    func picServer() {
        
    }
}
